ex2_js-basics-part2:

Indigoinspn opened the pull request:
task-01.js - Function processes the type of passed value.
task-02.js - Function iterates over the elements of given array and returns it's length.
task-03.js - Function counts and return odd, even and zero items.
task-04.js - Function returns result of check for uniformity of items.
task-05.js - Function returns the maximum value from items of given array.
task-06.js - Function returns result of check of taken value for simpleness if it not greater than 1000.

Indigoinspn committed:
ex2_js-basics-part2:

task-01.js - Function processes the type of passed value.
task-02.js - Function iterates over the elements of given array and returns it's length.
task-03.js - Function counts and return odd, even and zero items.
task-04.js - Function returns result of check for uniformity of items.
task-05.js - Function returns the maximum value from items of given array.
task-06.js - Function returns result of check of taken value for simpleness if it not greater than 1000.

aneelia commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,13 @@
+function processType(input) {
+    var outputType;
```
1. код можно и нужно упросить, return сам по себе вернет undefined, переменная не нужна
2. прочитай про NaN
3. typeof  принято писать без ()

aneelia commented on src/ex2_js-basics-part2/task-02.js:
```diff
@@ -0,0 +1,10 @@
+function loopObject (arr) {
+    var quantity = arr.length - 1;
```
нет смысла заводить переменную, просто по месту выполни операцию

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,21 @@
+function oddEvenZeroNum (arr) {
+    var index = arr.length - 1;
```
var лучше объявлять 1 раз, переменные перечислить через запятую

aneelia commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,13 @@
+function checkForRepeat (arr) {
+    var index = arr.length-2;
```
то же самое замечание

aneelia commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,15 @@
+function findMaxNum (arr) {
+    var index = arr.length-1;
```
var

aneelia replied:
и ненужная переменная

aneelia commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,28 @@
+    if (num < 2 || num > 1000) {
+        message = "Данные неверны" ; 
+    } else if (num === 2) {
+        message = "Число " + num + " - простое число"; 
```
не надо проверять 2, оно не особенное. надо проверять 1 и 0

aneelia commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,28 @@
+            } 
+            else {
+
+                if ( ++i === num) {                    
```
конкретно здесь не нужно никакого условия это последний else

а так же break, continue лишнее. можно не присваивать message, а делать return. код станет меньше

Indigoinspn committed:
ex2_js-basics-part2:
Improved code
task-01.js - Function processes the type of passed value.
task-02.js - Function iterates over the elements of given array and returns it's length.
task-03.js - Function counts and return odd, even and zero items.
task-04.js - Function returns result of check for uniformity of items.
task-05.js - Function returns the maximum value from items of given array.
task-06.js - Function returns result of check of taken value for simpleness if it not greater than 1000.

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,18 @@
+        } 
+    }
+
+    return ( [ even, odd, zero ] );   
```
можно без (). на будущее запомни
